cpp/modmesh/inout/gmsh.cpp

```diff
@@ -94,6 +94,159 @@ void Gmsh::build_interior(const std::shared_ptr<StaticMesh> & blk)
     blk->build_boundary();
     blk->build_ghost();
 }
+
+// Check the finite state machine state transition is valid or not to check msh file format is correct
+bool Gmsh::is_valid_transition(const std::string s)
+{
+    if (last_fmt_state == FormatState::BEGIN)
+    {
+        return s == "$MeshFormat";
+    }
+    else if (last_fmt_state == FormatState::META_END || last_fmt_state == FormatState::PYHSICAL_NAME_END)
+    {
+        return s == "$PhysicalNames" || s == "$Nodes";
+    }
+    else if (last_fmt_state == FormatState::NODE_END)
+    {
+        return s == "$Elements";
+    }
+
+    return false;
+}
+
+void Gmsh::load_meta(void)
+{
+    std::string line;
+
+    while (std::getline(stream, line) && line.find('$') == std::string::npos)
+    {
+        auto tokens = tokenize(line, ' ');
+        msh_ver = std::stod(tokens[0]);
+
+        // The parse only support ver 2.2 msh file.
+        if (msh_ver != 2.2)
```

Comment:
I will choose to use `constexpr` for the number.

Reply:
Why not just use a macro ?

Reply:
Macro does not allow compilers to check type. When macros and constexpr both work, choose constexpr.

Reply:
For version number, integer should be used.  Floating is not exact and should be avoided.